### ISSUE TITLE
Remapping bismark results to the genome

### ISSUE THREAD

**Issue by dgescher** [FelixKrueger/Bismark#519]
Hi Felix,
it seems like that bismark runs well, but I miss the mapping information in the reports. Actually, I'm not sure wether mapping information would be to expect in the bismark report file. There are some points I'm wondering about:
- in the graph within the bismark-report.html as well as within the other complete output .txt files I only get information about the first 150bp, but there must be data for the whole test-sequences somewhere.
- in the bismark endreportfile.html I don't get any information about mapping/localization to the genome (is that normal)?
- a colleague got other file names (but perhaps using an older version?), ending with -ready.bam and -ready.bam.bai
 Should I also expect to get them as an output?
- in the later performed multiqc I get an incomplete report (samtools and fastqc results are missing), especially lacking the mapping information). 

how it looks now (only graph of the multiqc-report): 
![image](https://user-images.githubusercontent.com/104357006/186441701-274a8f3b-9872-4571-98dd-e6d78017971f.png)

how it should look like (extract from the full report with mapping information):
![image](https://user-images.githubusercontent.com/104357006/186441887-4f7b53ed-cc43-417f-9117-afd45aaff066.png)

Besides of the issues above, how can I use the samtools and fastq features to get the full information of genome mapping of the methylated Cytosins?
Is it normal not to get the mapping information within only bismark (that would confirm my impression that bismark runs well, and perhaps it is a problem with later steps)?
Where (in which output file of bismark) can I find the results for the complete sample-sequence (not only 150 bp)?

Thanks so much for help!!
Best regards from RWTH Aachen, Germany
Dorothee

**Comment by FelixKrueger** [comment]
Hi Dorothee, 

I am currently traveling, but I will try to take a closer look upon my return (next weekend). Apologies for the delay.

**Comment by dgescher** [comment]
> Hi Dorothee,
> 
> I am currently traveling, but I will try to take a closer look upon my return (next weekend). Apologies for the delay.

Hi Felix, thank you for answering, no worries about the delay! Perhaps I only made simple mistakes but I'm "hanging" on this issue and don't understand how to use samtools or fastq in order to get a complete result from bismark. I'm a very beginner also in ubuntu and thus makes me feeling really helpless and stupid. If helpful, I also could send full data in a personal mail.  Best, Dorothee

**Comment by dgescher** [comment]
Hi Felix,
in between I better understood my data. The only current problem still is that Bismark doesn't find GD:Graph:lines in the methylationextractor step, even if we installed perl graph modules serveral times. Perhaps this hampers getting graphs. (reasons for not getting the intended graph (see question above) I meanwhile would assign to have not used samtools steps).
Have you heard about this Graph:lines problem? Best, Dorothee

**Comment by FelixKrueger** [comment]
Hi Dorothee,

Apologies for the late reply, I am only now catching up...

I imagine you are might be getting warning messages that the GD::Graph module is not installed, however this should not matter at all for processing at all. You should get a much nicer HTML report by completing all of the Bismark steps (alignment, deduplication, methylation extraction, followed by running `bismark2report` in the same folder that contains all of the output files and reports.

Regarding some of the questions you asked in the original post: 

- in the graph within the bismark-report.html as well as within the other complete output .txt files I only get information about the first 150bp, but there must be data for the whole test-sequences somewhere.
 >> I am not quite sure I understand the question here, the Bismark report will give you average statistics about the entire run, but does't contain any specific details about locations or sequences.

- in the bismark endreportfile.html I don't get any information about mapping/localization to the genome (is that normal)?
>>  Not sure I have ever heard of an enreportfile.html, where did you get this from?

- a colleague got other file names (but perhaps using an older version?), ending with -ready.bam and -ready.bam.bai
Should I also expect to get them as an output?
>> Bismark does not produce -ready.bam files or BAM indexes (.bai) thereof. This must have been produced using other packages, such as Samtools.

- in the later performed multiqc I get an incomplete report (samtools and fastqc results are missing), especially lacking the mapping information)
>> A MultiQC run FastQC and Bismark reports does not contain any mapping information (as in the locations), but it should show you a summary of the mapping, deduplication, and methylation extraction statistics as well as the M-bias plots.

You might want to take a look at the training material for methylation analysis here: https://www.bioinformatics.babraham.ac.uk/training.html#bsseq, this should give you a very good overview of the Bismark workflow with visualisation and analysis.

**Comment by dgescher** [comment]
Hi Felix,
thanks very much for your support. Unfortunately you are right that I created misunderstandings due to linguistic mixtures. 
- with "endreportfile.html" I tried to denote the .html-file that generates as a result from report2bismark (SAMPLE.html) in the end.
- the main confusion in my Write was to take same words for mapping and localizing, because in German it is similar, sorry. My question was, wether I can get information from the bismark output-files about the localizing in the genome (with which I did the mapping analyzes), and why it is not seen in the multiqc-report (mutiqc should take all informations from that folder including all bismark output files. Perhaps they must be un-gzipped?  But I've understood now that that includes not the localization within the genome. 
- thanks for the training material as well! That's very helpful.
